core/src/main/java/com/linecorp/armeria/client/AbstractDnsResolverBuilder.java

```diff
@@ -523,11 +523,11 @@ protected final Consumer<DnsNameResolverBuilder> buildConfigurator(EventLoopGrou
                    .searchDomains(ImmutableList.of())
                    .decodeIdn(decodeIdn);
 
-            if (queryTimeoutMillisForEachAttempt > 0) {
+            if (queryTimeoutMillisForEachAttempt > 0 && queryTimeoutMillisForEachAttempt < Long.MAX_VALUE) {
                 builder.queryTimeoutMillis(queryTimeoutMillisForEachAttempt);
             } else {
-                if (queryTimeoutMillis == 0) {
-                    builder.queryTimeoutMillis(Long.MAX_VALUE);
+                if (queryTimeoutMillis == 0 || queryTimeoutMillis == Long.MAX_VALUE) {
+                    builder.queryTimeoutMillis(-1);
```

Comment:
What I meant was setting 1 hour instead of `-1` when the `queryTimeoutMillis` is `MAX_VALUE` or greater than 1 hour.

Reply:
Understood.